src/cluster/mcl.cpp

```diff
@@ -0,0 +1,443 @@
+	// }
+	// 
+	// testing
+	// vector<Eigen::Triplet<float>> data;
+	// data.emplace_back(0, 0, 0.2f);
+	// data.emplace_back(0, 1, 0.25f);
+	// data.emplace_back(0, 5, 0.333f);
+	// data.emplace_back(0, 6, 0.25f);
+	// data.emplace_back(0, 9, 0.25f);
+	// data.emplace_back(1, 0, 0.20f);
+	// data.emplace_back(1, 1, 0.25f);
+	// data.emplace_back(1, 2, 0.25f);
+	// data.emplace_back(1, 4, 0.20f);
+	// data.emplace_back(2, 1, 0.25f);
+	// data.emplace_back(2, 2, 0.25f);
+	// data.emplace_back(2, 3, 0.20f);
+	// data.emplace_back(2, 4, 0.20f);
+	// data.emplace_back(3, 2, 0.25f);
+	// data.emplace_back(3, 3, 0.20f);
+	// data.emplace_back(3, 7, 0.20f);
+	// data.emplace_back(3, 8, 0.20f);
+	// data.emplace_back(3, 10, 0.20f);
+	// data.emplace_back(4, 1, 0.25f);
+	// data.emplace_back(4, 2, 0.25f);
+	// data.emplace_back(4, 4, 0.20f);
+	// data.emplace_back(4, 6, 0.25f);
+	// data.emplace_back(4, 7, 0.20f);
+	// data.emplace_back(5, 0, 0.20f);
+	// data.emplace_back(5, 5, 0.333f);
+	// data.emplace_back(5, 9, 0.25f);
+	// data.emplace_back(6, 0, 0.20f);
+	// data.emplace_back(6, 4, 0.20f);
+	// data.emplace_back(6, 6, 0.25f);
+	// data.emplace_back(6, 9, 0.25f);
+	// data.emplace_back(7, 3, 0.20f);
+	// data.emplace_back(7, 4, 0.20f);
+	// data.emplace_back(7, 7, 0.20f);
+	// data.emplace_back(7, 8, 0.20f);
+	// data.emplace_back(7, 10, 0.20f);
+	// data.emplace_back(8, 3, 0.20f);
+	// data.emplace_back(8, 7, 0.20f);
+	// data.emplace_back(8, 8, 0.20f);
+	// data.emplace_back(8, 10, 0.20f);
+	// data.emplace_back(8, 11, 0.333f);
+	// data.emplace_back(9, 0, 0.20f);
+	// data.emplace_back(9, 5, 0.333f);
+	// data.emplace_back(9, 6, 0.25f);
+	// data.emplace_back(9, 9, 0.25f);
+	// data.emplace_back(10, 3, 0.20f);
+	// data.emplace_back(10, 7, 0.20f);
+	// data.emplace_back(10, 8, 0.20f);
+	// data.emplace_back(10, 10, 0.20f);
+	// data.emplace_back(10, 11, 0.333f);
+	// data.emplace_back(11, 8, 0.20f);
+	// data.emplace_back(11, 10, 0.20f);
+	// data.emplace_back(11, 11, 0.333f);
+	// vector<unordered_set<uint32_t>> cr;
+	// float inf = 2.0;
+	// float exp = 2.0;
+	// if(false){
+	// 	Eigen::SparseMatrix<float> tmp(12,12);
+	// 	tmp.setFromTriplets(data.begin(), data.end());
+	// 	printMatrix(&tmp);
+	// 	cr = markov_process(&tmp, inf, exp);
+	// 	printMatrix(&tmp);
+	// }
+	// else{
+	// 	Eigen::MatrixXf tmp = Eigen::MatrixXf::Zero(12,12);
+	// 	for(Eigen::Triplet<float> const & t : data){
+	// 		tmp(t.row(), t.col()) = t.value();
+	// 	}
+	// 	printMatrix(&tmp);
+	// 	cr = markov_process(&tmp, inf, exp);
+	// 	printMatrix(&tmp);
+	// }
+	// cout << "Found "<<cr.size()<<" clusters"<<endl;
+	// for(auto& set: cr){
+	// 	cout<< "set :";
+	// 	for(auto item : set){
+	// 		cout<< " " << item;
+	// 	}
+	// 	cout<<endl;
+	// }
+	// return;
+
+	const size_t seq_count = db->ref_header.sequences;
+	statistics.reset();
+	config.command = Config::blastp;
+	config.no_self_hits = false;
+	config.output_format = { "clus" };
+	config.query_cover = 80;
```

Comment:
This should be configurable by the user.

Reply:
Which line does this comment apply to (this is possibly due to my github ignorance), 325? If so, I agree :)

Reply:
Yes, I meant the query and subject cover.

Reply:
Thanks. Done